internal/metrics/metrics.go

```diff
@@ -32,73 +32,94 @@ const (
 	metricLabelParentKind        = "parent_kind"
 	metricLabelParentNamespace   = "parent_namespace"
 	metricLabelParentName        = "parent_name"
-)
 
-var metricTemplateUsage = prometheus.NewGaugeVec(
-	prometheus.GaugeOpts{
-		Namespace: kcmv1.CoreKCMName,
-		Name:      "template_usage",
-		Help:      "Number of templates currently in use",
-	},
-	[]string{metricLabelTemplateKind, metricLabelTemplateName, metricLabelParentKind, metricLabelParentNamespace, metricLabelParentName},
+	metricLabelIPAMKind      = "ipam_kind"
+	metricLabelIPAMNamespace = "ipam_namespace"
+	metricLabelIPAMName      = "ipam_name"
 )
 
-var metricTemplateInvalidity = prometheus.NewGaugeVec(
-	prometheus.GaugeOpts{
-		Namespace: kcmv1.CoreKCMName,
-		Name:      "template_invalidity",
-		Help:      "Number of invalid templates",
-	},
-	[]string{metricLabelTemplateKind, metricLabelTemplateNamespace, metricLabelTemplateName},
+var (
+	metricTemplateUsage = newGaugeVec("template_usage", "Number of templates currently in use",
+		metricLabelTemplateKind, metricLabelTemplateName, metricLabelParentKind, metricLabelParentNamespace, metricLabelParentName)
+
+	metricTemplateInvalidity = newGaugeVec("template_invalidity", "Number of invalid templates",
+		metricLabelTemplateKind, metricLabelTemplateNamespace, metricLabelTemplateName)
+
+	metricIPAMClaimUse = newGaugeVec("ipam_claim_use", "Number of IPAM claims currently in use",
+		metricLabelIPAMKind, metricLabelIPAMNamespace, metricLabelIPAMName)
+
+	metricIPAMClaimsBound = newGaugeVec("ipam_claims_bound", "Number of IPAM claims which are bound",
+		metricLabelIPAMKind, metricLabelIPAMNamespace, metricLabelIPAMName)
 )
 
 func init() {
 	metrics.Registry.MustRegister(
 		metricTemplateUsage,
 		metricTemplateInvalidity,
+		metricIPAMClaimUse,
+		metricIPAMClaimsBound,
 	)
 }
 
-func TrackMetricTemplateUsage(ctx context.Context, templateKind, templateName, parentKind string, parent metav1.ObjectMeta, inUse bool) { //nolint:revive // false-positive
-	var value float64
-	if inUse {
+func newGaugeVec(name, help string, labels ...string) *prometheus.GaugeVec {
+	return prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: kcmv1.CoreKCMName,
+			Name:      name,
+			Help:      help,
+		},
+		labels,
+	)
+}
+
+func setGaugeAndLog(ctx context.Context, gauge *prometheus.GaugeVec, labels prometheus.Labels, active bool, logMsg string) { //nolint:revive
+	value := 0.0
+	if active {
 		value = 1
 	}
+	gauge.With(labels).Set(value)
+
+	ctrl.LoggerFrom(ctx).V(1).Info(logMsg, append(labelMapToSlice(labels), "value", value)...)
```

Comment:
to avoid calling `labelMapToSlice` when it is not necessary, it's better to check whether the level is enabled:

```o
l := ctrl.LoggerFrom(ctx)
if l.V(1).Enabled() { // log }
```